lang/ruby31/files/patch-thread_pthread.diff

```diff
@@ -0,0 +1,44 @@
+--- thread_pthread.c.orig	2022-04-12 19:11:15.000000000 +0800
++++ thread_pthread.c	2022-10-24 05:26:46.000000000 +0800
+@@ -40,6 +40,10 @@
+ #include <time.h>
+ #include <signal.h>
+ 
++#if defined __APPLE__
++# include <AvailabilityMacros.h>
++#endif
++
+ #if defined(HAVE_SYS_EVENTFD_H) && defined(HAVE_EVENTFD)
+ #  define USE_EVENTFD (1)
+ #  include <sys/eventfd.h>
+@@ -1732,9 +1736,29 @@
+     return INT2FIX(tid);
+ #elif defined(__APPLE__)
+     uint64_t tid;
+-    int e = pthread_threadid_np(target_th->thread_id, &tid);
++# if (!defined(MAC_OS_X_VERSION_10_6) || \
++      (MAC_OS_X_VERSION_MAX_ALLOWED < 1060) || \
++      defined(__POWERPC__) /* never defined for PowerPC platforms */)
++    const bool no_pthread_threadid_np = true;
++#   define NO_PTHREAD_MACH_THREAD_NP 1
++# elif MAC_OS_X_VERSION_MIN_REQUIRED >= 1060
++    const bool no_pthread_threadid_np = false;
++# else
++#   if !(defined(__has_attribute) && __has_attribute(availability))
++    /* __API_AVAILABLE macro does nothing on gcc */
++    __attribute__((weak)) int pthread_threadid_np(pthread_t, uint64_t*);
++#   endif
++    /* Check weakly linked symbol */
++    const bool no_pthread_threadid_np = !&pthread_threadid_np;
++# endif
++    if (no_pthread_threadid_np) {
++        return ULL2NUM(pthread_mach_thread_np(pthread_self()));
++    }
++# ifndef NO_PTHREAD_MACH_THREAD_NP
++    int e = pthread_threadid_np(target_th->nt->thread_id, &tid);
```

Comment:
compile failed here on intel macOS 12.6.

> In file included from thread.c:352:
> ./thread_pthread.c:1758:44: error: no member named 'nt' in 'struct rb_thread_struct'
>     int e = pthread_threadid_np(target_th->nt->thread_id, &tid);
>                                 ~~~~~~~~~  ^
> 1 error generated.

"target_th->nt->thread_id" would be "target_th->thread_id".
ruby-3.1.2 does not have rb_thread_struct->nt in  vm_core.h.

I confirmed ruby31 passed core tests (run `make TESTS=ruby test-all`) with change to "target_th->thread_id" on intel macOS 12.6.